zipkin-simple.js

```diff
@@ -75,22 +75,27 @@ function get_child (trace_data) {
 		return create_root_trace()
 	}
 
+	var time = generate_timestamp()
+
 	return {
 		trace_id: trace_data.trace_id,
 		parent_span_id: trace_data.span_id,
 		span_id: generate_id(),
 		sampled: trace_data.sampled,
-		timestamp: generate_timestamp()
+		timestamp: time,
+		duration: time - trace_data.timestamp
```

Comment:
one thing I'd be careful about is setting duration on spans not started in this process. Ex. if a span was resumed from headers, it shouldn't report timestamp or duration. Not sure this could be the case here or not.

http://zipkin.io/pages/instrumenting.html

+1 on sending timestamp and duration on spans started! it helps!